android/app/src/main/java/org/fossasia/openevent/core/track/session/TrackSessionsActivity.java

```diff
@@ -358,4 +361,11 @@ public void showSnackbar(BookmarkStatus bookmarkStatus) {
         SnackbarUtil.setSnackbarAction(this, snackbar, bookmarkStatus)
                 .show();
     }
+
+    @Override
+    public void itemOnClick(int adapterPosition, int layoutPosition) {
+        trackSessionsActivityViewModel.setListPosition(layoutPosition);
+        Intent intent = new Intent(this, SessionDetailActivity.class);
+        startActivity(trackSessionsActivityViewModel.openSessionDetail(sessions.get(adapterPosition), intent));
+    }
```

Comment:
Unfortunately, even after moving the click handling in the view, we still don't have any less complex ViewHolder. Still, it is progress enough. But don't pass adapterPosition and layoutPosition here. Pass the session which was clicked

Reply:
Most of the part is just setting views depending on the value, so we will have to make some helper class to handle the if-else part then